warehouse/warehouse/scripts/warehouse_verify_publication.py

```diff
@@ -326,14 +328,25 @@ def main():
         # print(f"DBG: p_len = {p_len}")
 
         # obtain data from ESGF search API
-        docs = paginate_query(project, {"master_id": dsid}, qtype="Dataset", fields="version,data_node")
+        facets = {"project": f"{project}", "master_id": dsid}
+        facets.update(add_facet)
+
+        docs = safe_search_esgf(facets, qtype="Dataset", fields="version,data_node")
+        if not docs:
+            reason = f"Dataset query returned empty docs"
+            log_message("error", f"dataset_id {dsid}: {reason}")
+            continue
```

Comment:
Same comment applies about raising error (e.g., `ValueError`), instead of `continue`.

Reply:
In the “warehouse_vertify_publication” app, it can be given a list of hundreds of datasets to attempt to verify.  Perhaps “error” is to strong – but even then it is an error it finds in the dataset status, and needs to continue processing the other datasets.